src/containers/teamPage/ducks/selectors.ts

```diff
@@ -0,0 +1,45 @@
+import { AsyncRequest, AsyncRequestKinds } from '../../../utils/asyncRequest';
+import { TeamResponse, TeamProps, GoalProps, GoalResponseJSON } from './types';
+
+export const teamResponseRequestToTeamProps = (
+  team: AsyncRequest<TeamResponse, any>,
+): TeamProps => {
+  if (team.kind === AsyncRequestKinds.Completed) {
+    return teamResponseToTeamProps(team.result);
+  }
+  return emptyTeam;
+};
+
+export const teamResponseToTeamProps = (team: TeamResponse): TeamProps => {
+  return {
+    id: team.id,
+    name: team.name,
+    bio: team.bio,
+    members: team.members,
+    goals: mapGoalResponseJSONToGoalProps(team.goals),
+  };
+};
+
+const mapGoalResponseJSONToGoalProps = (
+  goals: GoalResponseJSON[],
+): GoalProps[] => {
+  return goals.map((goal) => {
+    return {
+      ...goal,
+      id: goal.goalId,
+      startDate: new Date(goal.startDate),
+      completeBy: new Date(goal.completeBy),
+      completionDate: !!goal.completeBy ? new Date(goal.completeBy) : null,
+    };
+  });
+};
+
+// In case the team request has not been completed.
+// This is to prevent the TeamProps from being undefined
+const emptyTeam: TeamProps = {
```

Comment:
I think this is going to refer to one object, if we mutate it that means this will also be mutated. Consider making a function that returns a fresh object every time instead or freezing it?
```suggestion
const emptyTeam: () => TeamProps = () => ({
  id: 0,
  name: '',
  bio: '',
  members: [],
  goals: [],
});
```

Reply:
Yeah, good idea. Although, it may make more sense to alter wherever this is used to simply use these defaults locally when an undefined object is received.